controller/DeliveryMgmt.php

```diff
@@ -0,0 +1,157 @@
+<?php
```

Comment:
This is a lot done in a controller. How about extracting logic to a dependency?

Reply:
This logic is only http related. The form creations & creation of delivery are not in the controller

Reply:
Ok. Still wouldn't hurt to simplify the big method. But ok for now...